azure-pipelines-v3.yml

```diff
@@ -31,13 +31,23 @@ jobs:
       packageType: sdk
       version: $(dotnetVersion)
 
+  # dotnet test
+  - task: DotNetCoreCLI@2
+    displayName: dotnet restore
+    condition: and(succeeded(), ne(variables['warm-up'], 'true'))
+    inputs:
+      command: restore
+      feedsToUse: config
+      nugetConfigPath: NuGet.Config
+
+
   # dotnet test
   - task: DotNetCoreCLI@2
     displayName: dotnet test
     condition: and(succeeded(), ne(variables['warm-up'], 'true'))
     inputs:
       command: test
-      arguments:  -warnAsError -c Release /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura
+      arguments:  -warnAsError -c Release --no-restore /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura
```

Comment:
Looks like we need an additional `dotnet build -warnAsError -c Release` step to tell between build failure and test failure.